### PR TITLE
[Merged by Bors] - chore: replace last uses of `lambda` instead of `fun`

### PR DESCRIPTION
**Pull request by grunweg**
Split out from #15896.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Real/Sqrt.lean [hunk 1]
```diff
@@ -115,15 +115,14 @@ noncomputable def sqrt (x : ℝ) : ℝ :=
...
-/- quotient.lift_on x
-  (λ f, mk ⟨sqrt_aux f, (sqrt_aux_converges f).fst⟩)
-  (λ f g e, begin
-    rcases sqrt_aux_converges f with ⟨hf, x, x0, xf, xs⟩,
-    rcases sqrt_aux_converges g with ⟨hg, y, y0, yg, ys⟩,
-    refine xs.trans (eq.trans _ ys.symm),
-    rw [← @mul_self_inj_of_nonneg ℝ _ x y x0 y0, xf, yg],
-    congr' 1, exact quotient.sound e
-  end)-/
+/- Quotient.liftOn x
```
Comment (mattrobball) [inline Mathlib/Data/Real/Sqrt.lean:118]: What is this comment doing?

Comment (eric-wieser) [reply Mathlib/Data/Real/Sqrt.lean:118]: This is some partial code to make `Real.sqrt` computable, which perhaps lost some context that was present in Lean 3.

Comment (eric-wieser) [reply Mathlib/Data/Real/Sqrt.lean:118]: The context was a
```
/- TODO(Mario): finish the proof
theorem sqrt_aux_converges (f : cau_seq ℚ abs) : ∃ h x, 0 ≤ x ∧ x * x = max 0 (mk f) ∧
  mk ⟨sqrt_aux f, h⟩ = x :=
begin
  rcases sqrt_exists (le_max_left 0 (mk f)) with ⟨x, x0, hx⟩,
  suffices : ∃ h, mk ⟨sqrt_aux f, h⟩ = x,
  { exact this.imp (λ h e, ⟨x, x0, hx, e⟩) },
  apply of_near,

  rsuffices ⟨δ, δ0, hδ⟩ : ∃ δ > 0, ∀ i, abs (↑(sqrt_aux f i) - x) < δ / 2 ^ i,
  { intros }
end -/
```
comment a few lines above.

Comment (mattrobball) [reply Mathlib/Data/Real/Sqrt.lean:118]: Can we add that back?

Comment (mattrobball) [reply Mathlib/Data/Real/Sqrt.lean:118]: Or remove and make an issue?

Comment (eric-wieser) [reply Mathlib/Data/Real/Sqrt.lean:118]: I think it's left from #11909. Let's remove it, but not in this PR.

**Comment by mattrobball** [comment]
bors merge

File: Mathlib/Data/Real/Sqrt.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/Data/Real/Sqrt.lean:118]: #15976

**Comment by grunweg** [comment]
Thanks for the fast review!

**Pull request closed without merging**